src/cs-utils/logger.hpp

```diff
@@ -59,27 +58,30 @@
 ///   An example would be:
 ///     [I][csp-stars] Failed to open Tycho catalogue file: File "../main.dat" does not exits! Did
 ///                    you download all required datasets?
-namespace cs::utils::logger {
+namespace cs::utils {
 
-/// This creates the default logger for "cs-utils" and is called at startup by the main() method.
-CS_UTILS_EXPORT void init();
+/// This creates the default singleton logger for "cs-utils" when called for the first time and
+/// returns it.
+CS_UTILS_EXPORT std::shared_ptr<spdlog::logger> logger();
+
+/// This creates the default logger for vista and is called at startup by the main() method.
+CS_UTILS_EXPORT void initVistaLogger();
 
 /// This signal is emitted whenever a message is logged with spdlog. The first argument is the
 /// logger's name, the second the log level, the last argument is the message.
-CS_UTILS_EXPORT Signal<std::string, spdlog::level::level_enum, std::string> const& onMessage();
+CS_UTILS_EXPORT Signal<std::string, spdlog::level::level_enum, std::string> const& onLogMessage();
 
 /// Call this method once from your plugin in order to create a new logger. The given name will
 /// be shown together with the log level in each message. The logger will print to the console and
-/// store it's messages in a file called cosmoscout.log. You need to call
-/// spdlog::set_default_logger(logger); in order to be able to use the global spdlog::info() etc.
-/// methods.
+/// store it's messages in a file called cosmoscout.log. If you want to, you could store the
+/// returned logger in function static variable in order to create a singleton.
 CS_UTILS_EXPORT std::shared_ptr<spdlog::logger> createLogger(std::string const& name);
```

Comment:
Is a `unique_ptr` better here? We can return a reference from the `logger()` functions.